### PR TITLE
Fix 7.7.1b test images

### PR DESCRIPTION
**Pull request by Katka92**
### What does this PR do?
Hotfix for 7.7.1b. Those images were not pushed to the upstream test repo, so the downstream images are failing.

### CODE REVIEW

**Comment by ibuziuk** [comment]
@Katka92 `7.7.1b` images are available, I believe we need to remove hot-fixes with 7.7.1 / 7.7.1b versions from this PR:

- che-devfile-registry `7.7.1b` https://quay.io/repository/eclipse/che-devfile-registry?tab=tags&tag=7.7.1b
- che-plugin-registry `7.7.1b` https://quay.io/repository/eclipse/che-plugin-registry?tab=tags&tag=7.7.1b
- https://quay.io/repository/eclipse/che-e2e?tab=tags&tag=7.7.1b

**Comment by ibuziuk** [comment]
And btw PR-check is back here - https://github.com/redhat-developer/rh-che/pull/1772#issuecomment-576804692

File: .ci/cico_rhche_prcheck.sh [hunk 1]
```diff
@@ -43,7 +43,8 @@ CHE_VERSION=$(getVersionFromPom)
 export CHE_VERSION
 
 echo "Running ${JOB_NAME} PR: #${RH_PULL_REQUEST_ID}, build number #${BUILD_NUMBER} for che-version:${CHE_VERSION}"
-.ci/cico_build_deploy_test_rhche.sh
+#.ci/cico_build_deploy_test_rhche.sh
```
Comment (ibuziuk) [inline .ci/cico_rhche_prcheck.sh]: why this got commented?

File: .ci/functional_tests_utils.sh [hunk 2]
```diff
@@ -121,7 +121,11 @@ function archiveArtifacts() {
 
 function getVersionFromPom() {
   version=$(scl enable rh-maven33 "mvn -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn org.apache.maven.plugins:maven-help-plugin:evaluate -q -Dexpression=project.parent.version -DforceStdout")
-  echo $version
+  if [[ "$version" == "7.7.1b" ]]; then
```
Comment (ibuziuk) [inline .ci/functional_tests_utils.sh]: not needed anymore

File: .ci/functional_tests_utils.sh [hunk 3]
```diff
@@ -183,11 +187,23 @@ function getActiveToken() {
 function getVersionFromProdPreview() {
   token=$(getActiveToken)
   version=$(curl -s -X OPTIONS --header "Content-Type: application/json" --header "Authorization: Bearer ${token}" https://che.prod-preview.openshift.io/api/ | jq '.buildInfo')
-  echo ${version//\"/}
+  version=${version//\"/}
```
Comment (ibuziuk) [inline .ci/functional_tests_utils.sh]: not needed anymore

File: .ci/functional_tests_utils.sh [hunk 4]
```diff
@@ -183,11 +187,23 @@ function getActiveToken() {
...
+  fi
+
 }
 
 function getVersionFromProd() {
   token=$(getActiveToken)
   version=$(curl -s -X OPTIONS --header "Content-Type: application/json" --header "Authorization: Bearer ${token}" https://che.openshift.io/api/ | jq '.buildInfo')
-  echo ${version//\"/}
+  version=${version//\"/}
+  if [[ "$version" == "7.7.1b" ]]; then
```
Comment (ibuziuk) [inline .ci/functional_tests_utils.sh]: not needed anymore

File: dockerfiles/e2e-saas/Dockerfile [hunk 5]
```diff
@@ -5,7 +5,7 @@
 # http://www.eclipse.org/legal/epl-v10.html
 
 ARG TAG=latest
-FROM docker.io/eclipse/che-e2e:$TAG
+FROM quay.io/eclipse/che-e2e:$TAG
```
Comment (ibuziuk) [inline dockerfiles/e2e-saas/Dockerfile]: I suspect this is the only change that is required atm

File: e2e-saas/tests/JavaMaven.spec.ts [hunk 6]
```diff
@@ -1,176 +0,0 @@
...
-                console.warn(`After clicking to the "Terminal" -> "Run Task ..." the "Quick Open Container" has not been displayed, one more try`);
-
-                await topMenu.selectOption('Terminal', 'Run Task...');
-                await quickOpenContainer.waitContainer();
-            }
-        }
-
-        await quickOpenContainer.clickOnContainerItem(task);
-    }
-});
```
Comment (ibuziuk) [inline e2e-saas/tests/JavaMaven.spec.ts]: Is it just cleanup not related to the main issue with `7.7.1b` / PR, periodic checks

File: functional-tests/devscripts/run_tests.sh [hunk 7]
```diff
@@ -111,18 +111,22 @@ if [[ "$PR_CHECK_BUILD" == "true" ]]; then
...
 	else
     echo "Could not found RH-Che tests image with tag ${version}."
-    curl --silent -f -lSL https://index.docker.io/v1/repositories/eclipse/che-e2e/tags/${version} > /dev/null
-    RESULT=$?
-    if [[ $RESULT != 0 ]]; then
-      echo "Could not found Che test image with tag ${version}. Building own RH-Che image based on Che image with nightly tag."
-      docker build --build-arg TAG=nightly -t e2e_tests dockerfiles/e2e-saas
-      rhche_image=e2e_tests
-    else
+    echo "curl -X GET https://quay.io/api/v1/repository/eclipse/che-e2e/tag/7.7.1/images returns:"
```
Comment (ibuziuk) [inline functional-tests/devscripts/run_tests.sh]: looks like this change and below are not needed anymore

Comment (Katka92) [reply functional-tests/devscripts/run_tests.sh]: Debugging echoes removed.

File: e2e-saas/tests/JavaMaven.spec.ts (hunk 6 above)
Comment (Katka92) [reply e2e-saas/tests/JavaMaven.spec.ts]: Yes, this class is not needed anymore for the tests. Can I leave it here or should I create separate PR for it?

File: .ci/functional_tests_utils.sh (hunk 4 above)
Comment (Katka92) [reply .ci/functional_tests_utils.sh]: removed

File: .ci/functional_tests_utils.sh (hunk 3 above)
Comment (Katka92) [reply .ci/functional_tests_utils.sh]: removed

File: .ci/functional_tests_utils.sh (hunk 2 above)
Comment (Katka92) [reply .ci/functional_tests_utils.sh]: removed

File: .ci/cico_rhche_prcheck.sh (hunk 1 above)
Comment (Katka92) [reply .ci/cico_rhche_prcheck.sh]: To speed up the test to the part we added, to see if it works. Removed now.

File: e2e-saas/tests/JavaMaven.spec.ts (hunk 6 above)
Comment (ibuziuk) [reply e2e-saas/tests/JavaMaven.spec.ts]: yeah, let's leave it here

File: functional-tests/devscripts/run_tests.sh [hunk 8]
```diff
@@ -111,16 +111,14 @@ if [[ "$PR_CHECK_BUILD" == "true" ]]; then
...
 	else
     echo "Could not found RH-Che tests image with tag ${version}."
-    curl --silent -f -lSL https://index.docker.io/v1/repositories/eclipse/che-e2e/tags/${version} > /dev/null
-    RESULT=$?
-    if [[ $RESULT != 0 ]]; then
-      echo "Could not found Che test image with tag ${version}. Building own RH-Che image based on Che image with nightly tag."
-      docker build --build-arg TAG=nightly -t e2e_tests dockerfiles/e2e-saas
-      rhche_image=e2e_tests
-    else
+    if [ -n $(curl -X GET https://quay.io/api/v1/repository/eclipse/che-e2e/tag/7.7.1/images | jq .status) ]; then
```
Comment (ibuziuk) [inline functional-tests/devscripts/run_tests.sh]: do we really need to handle 7.7.1 case ?

Comment (Katka92) [reply functional-tests/devscripts/run_tests.sh]: Damn, it should be ${version} here. I'll fix that, thank you very much for your review!

**Pull request merged**